src/wallet/wallet.ts

```diff
@@ -103,170 +142,442 @@ class HathorWalletServiceWallet extends EventEmitter {
         if (data && data.error === 'wallet-already-loaded') {
           // If it was already loaded, we have to check if it's ready
           await handleCreate(data.status);
+        } else {
+          throw new WalletRequestError('Error creating wallet.');
         }
+      } else {
+        throw new WalletRequestError('Error creating wallet.');
       }
-      // TODO Should we handle another error?
     }
   }
 
+  /**
+   * Return wallet auth token
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  getAuthToken(): string | null {
+    return this.authToken;
+  }
+
   /**
    * When the wallet starts, it might take some seconds for the wallet service to completely load all addresses
    * This method is responsible for polling the wallet status until it's ready
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
   async startPollingStatus() {
     try {
-      const res = await walletApi.getWalletStatus(this.walletId!);
+      const res = await walletApi.getWalletStatus(this);
       const data = res.data;
       if (res.status === 200 && data.success) {
         if (data.status.status === 'creating') {
           setTimeout(async () => {
```

Comment:
I feel this structure is too fragile because the pooling won't be called again if an exception occurs. I feel we should use a `setInterval(...)` that must be cancelled after the wallet is ready.

Reply:
Makes sense. Did the refactor